components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/AccessTokenDAOImpl.java

```diff
@@ -551,6 +568,15 @@ public AccessTokenDO getLatestAccessToken(String consumerKey, AuthenticatedUser
                     accessTokenDO.setTokenState(tokenState);
                     accessTokenDO.setTokenId(tokenId);
                     accessTokenDO.setGrantType(grantType);
+
+                    if (StringUtils.isNotEmpty(authzUser.getAccessingOrganization())) {
+                        accessTokenDO.getAuthzUser().setAccessingOrganization(authzUser.getAccessingOrganization());
```

Comment:
We can use `authzUser.getAccessingOrganization()` or we can fetch the data received from the DB. But ideally both are same as the received query was received by that value. Otherwise have to add another attribute for the select query.